h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -227,12 +244,19 @@ public void onBuildSuccess(final Model finishedModel, final ParallelModelBuilder
     public void onBuildFailure(final ParallelModelBuilder.ModelBuildFailure modelBuildFailure,
                                final ParallelModelBuilder parallelModelBuilder) {
       parallelSearchGridLock.lock();
+      Throwable ex = modelBuildFailure.getThrowable();
       try {
-        grid.appendFailedModelParameters(null, modelBuildFailure.getParameters(), modelBuildFailure.getThrowable());
+        grid.appendFailedModelParameters(null, modelBuildFailure.getParameters(), ex);
       } finally {
         parallelSearchGridLock.unlock();
       }
-      attemptBuildNextModel(parallelModelBuilder, null);
+      if (!Grid.isJobCanceled(ex) 
+              && grid.getModelCount() == 0 
+              && _consecutiveModelFailures.incrementAndGet() >= _maxConsecutiveModelFailures) {
```

Comment:
I feel like we should be able to get rid of this counter since we introduced "&& grid.getModelCount() == 0 "

the reason why I think it is an issue is that it introduces a bug for "recovered" grids - that is grids that are reloaded from disk and the building continues - the state is lost - or at least it seems like it could be lost

because we have model count == 0 we can just check how many parameters failed - `SearchFailures getFailures ` and see how many failed paramters are there

this would simplify the code, get rid off state and get me some sleep at night ;)

Reply:
good point! I forgot that we had access to the history of failures, will change this and care of your sleep patterns ;)

Reply:
I don't know why `GridSearch` API is so closed but it's a real pita to configure it from AutoML with it's 200 `startGridSearch` methods as confusing as each other.
Thinking about making a builder quickly, wdyt?

Reply:
An additive change like make a lot of sense